rust_src/src/fonts.rs

```diff
@@ -112,4 +113,17 @@ pub fn fontp(object: LispObject, extra_type: LispObject) -> bool {
     })
 }
 
+/// Return t if and only if font-spec SPEC matches with FONT.  FONT is a font-spec, font-entity,
+/// or font-object.
+#[lisp_fn]
+pub fn font_match_p(spec: LispObject, font: LispObject) -> bool {
```

Comment:
We could create `LispFontSpec` and `LispFontSomething`. I wrote 'something' since it is a font related object not necessarily a font.

Reply:
Might not be worth it. That depends on how many uses there are.

Reply:
Something similar already exists in that file. Can you clarify how your suggestion is different from what's there?

Reply:
if it exists, use it :-)

Reply:
Yeah, I'll work on the `From` impls there. The structs are a bit different than the ones elsewhere in Remacs which I didn't pick up on yesterday.